src/basilisp/core.lpy

```diff
@@ -5806,45 +5806,47 @@
-                           sym        (symbol kw-name)
-                           kw         (if kw-ns
-                                        (keyword kw-ns kw-name)
-                                        (keyword kw-name))
-                           or-binding (get ors sym)]
-                       (if or-binding
-                         [sym `(or (get ~fn-arg ~kw) ~or-binding)]
+                     (let [kw-ns   (namespace arg)
+                           kw-name (name arg)
+                           sym     (symbol kw-name)
+                           kw      (if kw-ns
+                                     (keyword kw-ns kw-name)
+                                     (keyword kw-name))]
+                       (if (contains? ors sym)
+                         [sym `(get ~fn-arg ~kw ~(get ors sym))]
                          [sym `(get ~fn-arg ~kw)])))
 
         map-binding (fn [f arg]
-                      (let [k          (f arg)
-                            or-binding (get ors arg)]
-                        (if or-binding
-                          [arg `(or (get ~fn-arg ~k) ~or-binding)]
+                      (let [k (f arg)]
+                        (if (contains? ors arg)
+                          [arg `(get ~fn-arg ~k ~(get ors arg))]
                           [arg `(get ~fn-arg ~k)])))
 
         sym-binding (fn [arg]
-                      (let [sym-name   (name arg)
-                            sym        (symbol sym-name)
-                            or-binding (get ors sym)]
-                        (if or-binding
-                          [sym `(or (get ~fn-arg (quote ~arg)) ~or-binding)]
+                      (let [sym-name (name arg)
+                            sym      (symbol sym-name)]
+                        (if (contains? ors sym)
+                          [sym `(get ~fn-arg (quote ~arg) ~(get ors sym))]
                           [sym `(get ~fn-arg (quote ~arg))])))
 
         named-binding (fn [arg]
-                        (let [binding    (get-in arg [:binding :name])
-                              key        (:key arg)
-                              or-binding (get ors binding)]
-                          (if or-binding
-                            [binding `(or (get ~fn-arg (quote ~key)) ~or-binding)]
+                        (let [binding (get-in arg [:binding :name])
+                              key     (:key arg)]
+                          (if (contains? ors binding)
+                            [binding `(get ~fn-arg (quote ~key) ~(get ors binding))]
                             [binding `(get ~fn-arg ~key)])))
 
         child-binding (fn [child]
                         (let [arg (get-in child [:binding :name])
                               k   (:key child)]
                           [arg `(get ~fn-arg ~k)]))]
     (concat
+     [fn-arg
+      `(if (seq? ~fn-arg)
+         (if (next ~fn-arg)
+           (apply hash-map ~fn-arg)
+           (first ~fn-arg))
+         ~fn-arg)]
```

Comment:
Clojure apparently takes sequences and coerces them to maps for map destructuring or, if the sequence is only a single element, takes the first arg as if it were a map.